src/Stryker.Core/Stryker.Core/MutantFilters/MutantFilterFactory.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Stryker.Core.DiffProviders;
+using Stryker.Core.Options;
+using System;
+using System.Collections.Generic;
+
+namespace Stryker.Core.MutantFilters
+{
+    public static class MutantFilterFactory
+    {
+        public static IDiffProvider DiffProvider { get; private set; }
+
+        public static IMutantFilter Create(StrykerOptions options)
+        {
+            if(options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            return new BroadcastMutantFilter(DetermineEnabledMutantFilters(options));
+        }
+
+        private static IEnumerable<IMutantFilter> DetermineEnabledMutantFilters(StrykerOptions options)
+        {
+            var enabledFilters = new List<IMutantFilter> {
+                    new FilePatternMutantFilter(),
+                    new IgnoredMethodMutantFilter(),
+                    new ExcludeMutationMutantFilter(),
+                    new ExcludeFromCodeCoverageFilter()
+                };
+
+            if (options.DiffEnabled)
+            {
+                enabledFilters.Add(new DiffMutantFilter(DiffProvider ?? new GitDiffProvider(options)));
+            }
+            
+
+            return enabledFilters;
+
+           
+        }
+
+        public static void SetDiffProvider(IDiffProvider diffProvider)
```

Comment:
What's the intended use of this method?

Reply:
I used this method for mocking of the `diffProvider` in de `DiffMutantFilter`. Now that I am rethinking this it could also be a nullable parameter of the `Create` method.

 The `diffProvider` has to be mocked because the constructor of `DiffMutantFilter` calls the `diffProvider` which in turn calls `Gitlib2sharp`. This results in an error in pipelines because these builds do a shallow fetch where branches don't exist.